packages/turbo-codemod/src/transforms/transform-literals-to-wildcards.ts

```diff
@@ -0,0 +1,124 @@
+import path from "path";
+import fs from "fs-extra";
+import { getTurboConfigs } from "@turbo/utils";
+import type { EnvWildcard, Schema as TurboJsonSchema } from "@turbo/types";
+
+import type { TransformerArgs } from "../types";
+import getTransformerHelpers from "../utils/getTransformerHelpers";
+import { TransformerResults } from "../runner";
+import { RootSchema } from "@turbo/types/src/types/config";
+
+// transformer details
+const TRANSFORMER = "transform-literals-to-wildcards";
+const DESCRIPTION = "Rewrite env fields to distinguish wildcards from literals";
+const INTRODUCED_IN = "1.10.0";
+
+// Rewriting of environment variable names.
+const asteriskLiteral = new RegExp("\\*", "g");
+function transformEnvVarName(envVarName: string): EnvWildcard {
+  let output = envVarName;
+
+  // Transform leading !
+  if (envVarName[0] === "!") {
+    output = `\\${output}`;
+  }
+
+  // Transform literal asterisks
+  output = output.replace(asteriskLiteral, "\\*");
+
+  return output;
+}
+
+function migrateRootConfig(config: RootSchema) {
+  let { globalEnv, globalPassThroughEnv } = config;
+
+  if (Array.isArray(globalEnv)) {
+    config.globalEnv = globalEnv.map(transformEnvVarName);
+  }
+  if (Array.isArray(globalPassThroughEnv)) {
+    config.globalPassThroughEnv = globalPassThroughEnv.map(transformEnvVarName);
+  }
+
+  return migrateTaskConfigs(config);
+}
+
+function migrateTaskConfigs(config: TurboJsonSchema) {
+  for (const [_, taskDef] of Object.entries(config.pipeline)) {
+    let { env, passThroughEnv } = taskDef;
```

Comment:
Could pull this list of known env var fields

```js
for const field of knownEnvFields {
  taskDef[field] = taskDef[field].map(transformEnvVarName)
}
```

Reply:
This was a "keep with the pattern" since we do exactly this in four other codemods. 🤷‍♂️